tracer/src/Datadog.Trace/Generated/net461/Datadog.Trace.SourceGenerators/Datadog.Trace.SourceGenerators.TelemetryMetric.TelemetryMetricGenerator/GaugeExtensions.g.cs

```diff
@@ -0,0 +1,67 @@
+﻿// <auto-generated/>
+#nullable enable
+
+namespace Datadog.Trace.Telemetry.Metrics;
+internal static partial class GaugeExtensions
+{
+    /// <summary>
+    /// The number of members in the enum.
+    /// </summary>
+    public const int Length = 4;
+
+    /// <summary>
+    /// Gets the metric name for the provided metric
+    /// </summary>
+    /// <param name="metric">The metric to get the name for</param>
+    /// <returns>The datadog metric name</returns>
+    public static string GetName(this Datadog.Trace.Telemetry.Metrics.Gauge metric)
+        => metric switch
+        {
+            Datadog.Trace.Telemetry.Metrics.Gauge.StatsBuckets => "stats_buckets",
```

Comment:
No, you can't generate non-c# files with source generators. Obviously we could create a Nuke target to generate this, the same way the `span_metadata.md` is, but I'd ask why? It would basically just be a copy of the `Gauge`/`Count` enums. It wouldn't guarantee that we _actually_ emit them either (just that they're defined). Not sure what the benefit of doing that work would be tbh

Reply:
> No, you can't generate non-c# files with source generators

Who would have assumed that something called source generators would generate only source code?

> but I'd ask why?

I don't know. It felt like a good idea at the time :D. More seriously, my idea would be to have something declarative about what we can generate. Could be a first step to make sure all tracers try to emit the same metrics without reconciliation/normalization on the backend side. But it's a more global discussion and maybe not the best solution

Reply:
> Who would have assumed that something called source generators would generate only source code?

To be fair, they're planning on supporting that in future 😉 

> Could be a first step to make sure all tracers try to emit the same metrics without reconciliation/normalization on the backend side

To be clear, the telemetry intake is going to enforce the allowed metrics/tags. 

> my idea would be to have something declarative

My argument is that the enums are pretty declarative already 😉